Userland/Libraries/LibCore/Environment.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2021-2022, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2021-2022, Kenneth Myhra <kennethmyhra@serenityos.org>
+ * Copyright (c) 2021-2024, Sam Atkins <atkinssj@serenityos.org>
+ * Copyright (c) 2022, Matthias Zimmerman <matthias291999@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Environment.h"
+#include <AK/ByteString.h>
+
+#if defined(AK_OS_MACOS)
+#    include <crt_externs.h>
+#else
+extern char** environ;
+#endif
+
+namespace Core::Environment {
+
+char** raw_environ()
+{
+#if defined(AK_OS_MACOS)
+    return *_NSGetEnviron();
+#else
+    return environ;
+#endif
+}
+
+static Entry entry_from_string(char const* raw_input)
+{
+    StringView input { raw_input, strlen(raw_input) };
+    auto split_index = input.find('=');
+    if (!split_index.has_value()) {
+        return Entry {
+            .full_entry = input,
+            .name = input,
+            .value = ""sv,
+        };
+    }
+
+    return Entry {
+        .full_entry = input,
+        .name = input.substring_view(0, *split_index),
+        .value = input.substring_view(*split_index + 1),
+    };
+}
+
+void for_each_entry(Function<IterationDecision(Entry&)> callback)
+{
+    auto environment = raw_environ();
+    for (size_t i = 0; environment[i]; ++i) {
+        auto entry = entry_from_string(environment[i]);
+        if (callback(entry) == IterationDecision::Break)
+            break;
+    }
+}
+
+size_t size()
+{
+    auto environment = raw_environ();
+
+    size_t environ_size = 0;
+    while (environment[environ_size])
+        ++environ_size;
+
+    return environ_size;
+}
+
+Vector<Entry> entries()
+{
+    Vector<Entry> entries;
+
+    for_each_entry([&entries](Entry& entry) {
+        entries.append(entry);
+        return IterationDecision::Continue;
+    });
+
+    return entries;
+}
+
+Optional<StringView> get(StringView name)
+{
+    auto* result = ::getenv(name.to_byte_string().characters());
+    if (result)
+        return StringView { result, strlen(result) };
+    return {};
+}
+
+Optional<StringView> get_secure(StringView name)
+{
+    auto* result = ::secure_getenv(name.to_byte_string().characters());
```

Comment:
Same here, though a call to `getauxval(AT_SECURE)` will be required.

Reply:
Same StringBuilder trick applies here.